hummingbot/core/utils/api_throttler.py

```diff
@@ -0,0 +1,170 @@
+import asyncio
+import time
+
+from collections import deque
+from dataclasses import dataclass
+from enum import Enum
+from typing import (
+    Deque,
+    Dict,
+    List,
+)
+
+
+class RateLimitType(Enum):
+    FIXED = 1
+    WEIGHTED = 2
+    PER_METHOD = 3
+
+
+@dataclass
+class RateLimit():
+    limit: int
+    time_interval: float
+    path_url: str = ""
+    weight: int = 1
+
+
+@dataclass
+class TaskLog():
+    timestamp: float
+    path_url: str = ""
+    weight: int = 1
+
+
+Limit = int             # Integer representing the no. of requests be time interval
+RequestPath = str       # String representing the request path url
+RequestWeight = int     # Integer representing the request weight of the path url
+
+Seconds = float
+
+
+class APIThrottler:
```

Comment:
Is this still required now that we have APIThrottlerBase hierarchy?

Reply:
Yes this will be removed